File: Q3/antlrJava/antlrJava/src/main/java/com/mycompany/antlrjava/Selector.java

import java.io.File;
import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;

public class Selector{ 
    private int number; 
    private String attribute; 
    private String path; 




    Selector(int number, String attribute, String path){ 
        this.number = number; 
        this.attribute = attribute;
        this.path = path; 
    }


    if(this.attribute === "fspOldest"){ 
        this.fspOldest(false); 
    }else if(this.attribute === "fspNewest"){ 
        this.fspNewest();
    }else if(this.attribute === "fspBiggest"){ 
        fspBiggest(false); 
    }else if(this.attribute === "fspSmallest"){ 
        fspSmallest(); 
    }






    public void fspOldest(Boolean reverseStatus){ 
        final List<File> files = Arrays.asList(new File(this.path).listFiles()); 
        final Map<File, Long> modified = new HashMap<File, Long>(); 
        for (final File file : files){ 
            modified.put(file, file.lastModified()); 
        }

        Collections sortedFile = sortingFile(files, reverseStatus, modified); 
        if(this.sortedFile.length() < this.attribute){ 
            this.attribute = this.sortedFile.length(); 
        }
        for(int i = 0, i < this.attribute; i++){ 
            System.out.println(sortedFile[i]); 
        }


    }

    public void fspNewest(){ 
        this.fspOldest(true)
    }
    private Collections sortingFile(File file, Boolean reverseStatus, Map<File, Long> modified){
        if(reverseStatus == false){ 
            return Collections.sort(files, new Comparator<File>(){ 
                @override
                public int compare(final File f1, final File f2){ 
                    return modified.get(f1).compare(modified.get(f2))
                }
            }); 
        }else if(reverseStatus == true){ 
            return Collections.sort(files, new Comparator<File>(){ 
                @override
                public int compare(final File f1, final File f2){ 
                    return modified.get(f1).compare(modified.get(f2))
                }
            }, Collections.reverseOrder()); 
        } 

        return null; 


        
    }




    public void fspBiggest(Boolean reverseStatus){ 
        File files = new File(this.path); 
        Collections sortedSizeFile = sortingFileSize(files, reverseStatus);
        if(this.sortedSizeFile.length() < this.attribute){ 
            this.attribute = this.sortedFile.length(); 
        }
        for(int i = 0, i < this.attribute; i++){ 
            System.out.println(sortedSizeFile[i]); 
        }
    }

    public void fspSmallest(){ 
        fspBiggest(true)

    }

   private Collections sortingFileSize(File file, Boolean reverseStatus){
        if(reverseStatus == false){ 
            return Collections.sort(files); 
        }else if(reverseStatus == true){ 
            return Collections.sort(files,Collections.reverseOrder()); 
        } 

        return null; 


        
    }

}

File: Q2/fspow/fspow/Selector.py
import time 
from pathlib import Path, PosixPath 
import pathlib
import os 
import glob
import operator
from datetime import datetime


class Selector:
    number = None 
    attribute = None
    path = None 

    def __init__(self, path:str="",number:int=0, attribute:str=""): 
        self.path = Path(path) #Creating a path based on what inputted.
        self.number = number
        self.attribute = attribute


        if(attribute == "fspOldest"):
            print("\nOldest File First:")
            self.fspOldest()
        elif(attribute == "fspNewest"): 
            print("\nOldest File First:")
            self.fspNewset()
        elif(attribute =="fspBiggest"): 
            print("\nBiggest file First:")
            self.fspBiggest()
        elif(attribute =="fspSmallest"): 
            print("\nSmallest file First:")
            self.fspSmallest()
        

    def fspOldest(self, reverseStatus=False): #Oldest to newest
        #Get all the files
        listOfFiles = filter( lambda x: os.path.isfile(os.path.join(self.path, x)),os.listdir(self.path) )
        #Sort them out reverse= based on order
        listOfFiles = sorted( listOfFiles,key = lambda x: os.path.getmtime(os.path.join(self.path, x)),reverse=reverseStatus)

        print("[+]Files count in folder: ", len(listOfFiles), "below are ", self.number , "files")

        if(len(listOfFiles) < self.number): # Avoid OUTOFINDEXERROR
            self.number = len(listOfFiles)

        #print the dates with files name 
        for i in range(0,self.number): 
            filePath = os.path.join(self.path, listOfFiles[i])
            timeStamp = time.strftime('%a %b %d %H:%M:%S %Y',time.gmtime(os.path.getmtime(filePath)))
            print(timeStamp, ' = ', listOfFiles[i])    

    def fspNewset(self,): #Newest to oldest
        self.fspOldest(True)


    def fspBiggest(self, reverseStatus=True): 
        listOfFiles = filter( lambda x: os.path.isfile(os.path.join(self.path, x)),os.listdir(self.path) )
        listOfFiles = sorted( listOfFiles,key = lambda x: os.path.getsize(os.path.join(self.path, x)),reverse=reverseStatus)
        print("[+]Files count in folder: ", len(listOfFiles), "below are ", self.number , "files")

        if(len(listOfFiles) < self.number): # Avoid OUTOFINDEXERROR
            self.number = len(listOfFiles)
        
        #print the dates with files name 
        for i in range(0,self.number): 
            filePath = os.path.join(self.path, listOfFiles[i])
            timeStamp = os.path.getsize(filePath)
            print(round(int(timeStamp)/1024, 2), " MB ", ' = ', listOfFiles[i])    

    def fspSmallest(self,): 
        self.fspBiggest(reverseStatus=False) 




